src/main/java/com/o19s/es/ltr/action/TransportClearCachesAction.java

```diff
@@ -0,0 +1,124 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.o19s.es.ltr.action;
+
+import com.o19s.es.ltr.action.ClearCachesAction.ClearCachesNodeResponse;
+import com.o19s.es.ltr.action.ClearCachesAction.ClearCachesNodesRequest;
+import com.o19s.es.ltr.action.ClearCachesAction.ClearCachesNodesResponse;
+import com.o19s.es.ltr.feature.store.index.Caches;
+import org.elasticsearch.action.FailedNodeException;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.nodes.BaseNodeRequest;
+import org.elasticsearch.action.support.nodes.TransportNodesAction;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+import java.io.IOException;
+import java.util.List;
+
+public class TransportClearCachesAction extends TransportNodesAction<ClearCachesNodesRequest, ClearCachesNodesResponse,
+        TransportClearCachesAction.ClearCachesNodeRequest, ClearCachesNodeResponse> {
+    private final Caches caches;
+
+    @Inject
+    public TransportClearCachesAction(Settings settings, ThreadPool threadPool,
+                                         ClusterService clusterService, TransportService transportService,
+                                         ActionFilters actionFilters, IndexNameExpressionResolver indexNameExpressionResolver,
+                                         Caches caches) {
+        super(settings, ClearCachesAction.NAME, threadPool, clusterService, transportService, actionFilters, indexNameExpressionResolver,
+                ClearCachesNodesRequest::new, ClearCachesNodeRequest::new, ThreadPool.Names.MANAGEMENT, ClearCachesNodeResponse.class);
+        this.caches = caches;
+    }
+
+    @Override
+    protected ClearCachesNodesResponse newResponse(ClearCachesNodesRequest request, List<ClearCachesNodeResponse> responses,
+                                                   List<FailedNodeException> failures) {
+        return new ClearCachesNodesResponse(clusterService.getClusterName(), responses, failures);
+    }
+
+    @Override
+    protected ClearCachesNodeRequest newNodeRequest(String nodeId, ClearCachesNodesRequest request) {
+        return new ClearCachesNodeRequest(nodeId, request);
+    }
+
+    @Override
+    protected ClearCachesNodeResponse newNodeResponse() {
+        return new ClearCachesNodeResponse();
+    }
+
+    @Override
+    protected ClearCachesNodeResponse nodeOperation(ClearCachesNodeRequest request) {
+        ClearCachesNodesRequest r = request.request;
+        switch (r.getOperation()) {
+        case ClearStore: {
```

Comment:
Code blocks in switch statements? Syntactically correct, but I know only one person who uses code blocks... (well, now two).